components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ApiProductsApiServiceImpl.java

```diff
@@ -726,77 +676,65 @@ public Response apiProductsGet(Integer limit, Integer offset, String query, Stri
-        String provider = null;
-        try {
-            APIProvider apiProvider = RestApiCommonUtil.getLoggedInUserProvider();
-            String username = RestApiCommonUtil.getLoggedInUsername();
-            // if not add product
-            provider = body.getProvider();
-            String context = body.getContext();
-            if (!StringUtils.isBlank(provider) && !provider.equals(username)) {
-                if (!APIUtil.hasPermission(username, Permissions.APIM_ADMIN)) {
-                    if (log.isDebugEnabled()) {
-                        log.debug("User " + username + " does not have admin permission ("
-                                + Permissions.APIM_ADMIN + ") hence provider (" + provider
-                                + ") overridden with current user (" + username + ")");
-                    }
-                    provider = username;
-                }
-            } else {
-                // Set username in case provider is null or empty
-                provider = username;
-            }
-
-            List<String> tiersFromDTO = body.getPolicies();
-            Set<Tier> definedTiers = apiProvider.getTiers();
-            List<String> invalidTiers = RestApiUtil.getInvalidTierNames(definedTiers, tiersFromDTO);
-            if (!invalidTiers.isEmpty()) {
-                RestApiUtil.handleBadRequest(
-                        "Specified tier(s) " + Arrays.toString(invalidTiers.toArray()) + " are invalid", log);
-            }
-            if (body.getAdditionalProperties() != null) {
-                String errorMessage = RestApiPublisherUtils
-                        .validateAdditionalProperties(body.getAdditionalProperties());
-                if (!errorMessage.isEmpty()) {
-                    RestApiUtil.handleBadRequest(errorMessage, log);
-                }
-            }
-            if (body.getVisibility() == null) {
-                //set the default visibility to PUBLIC
-                body.setVisibility(VisibilityEnum.PUBLIC);
-            }
+    /**
+     * Import an API Product by uploading an archive file. All relevant API Product data will be included upon the creation of
+     * the API Product. Depending on the choice of the user, provider of the imported API Product will be preserved or modified.
+     *
+     * @param fileInputStream       UploadedInputStream input stream from the REST request
+     * @param fileDetail            File details as Attachment
+     * @param preserveProvider      User choice to keep or replace the API Product provider
+     * @param importAPIs            Whether to import the dependent APIs or not.
+     * @param overwriteAPIProduct   Whether to update the API Product or not. This is used when updating already existing API Products.
+     * @param overwriteAPIs         Whether to update the dependent APIs or not. This is used when updating already existing dependent APIs of an API Product.
+     * @return API Product import response
```

Comment:
Missing @param and @throws

Reply:
Fixed via https://github.com/wso2/carbon-apimgt/commit/eb9b7181c0f6cb5b9624602d91d28a7479bf551b